### DISCUSSION TITLE
Why skipStories has been deprecated?

### DISCUSSION THREAD

**Discussion by sandbox4013** [oblador/loki#331] [General]
Eg: I need to skip tests on some devices, because some of pages available only for desktop
It's annoying to skip every story by loki.skip, but in configuration it's very handy!

So! Why did you choose to skip this option?

**Comment by techeverri** [comment]
There are a couple of alternatives to `skipStories` if you only want Loki to match certain stories based on a filter

You can use the property `storiesFilter` in the Loki config
🔗 https://loki.js.org/configuration.html#configurations

Or the command line argument `--storiesFilter` 
🔗 https://loki.js.org/command-line-arguments.html

**Reply by d3vl1m3** [reply]
I've looked at the docs for this and I don't understand the terminology used to set this up. 

What is actually required in the similar `storiesFilter` param `${kind} ${story}`? I can't find any examples of this.

I've noticed this ambiguity a few places in the documentation where something is stated but no hard examples are given for config file usage or how to use as a command line arg.

**Reply by crisheiss** [reply]
@techeverri Hi Tomás, is there any place were I can see an example of the `storiesFilter` being used? We some files that squash all stories of the component on a single story (so we don't end up with too many snapshots). I only see in the documentation that the expected type is string, I tried a couple of things but no result.
Thanks!

**Reply by Devwulf** [reply]
So apparently, the "kind" is the `title` property assigned in the Story meta object (like Meta<typeof SomeComponent>). And "story" is the name of the Story object. So if you have this kind of story:
```
const meta: Meta<typeof SomeComponent> = {
    title: "Some/Category/To/SomeComponent",
    ...
};

type Story = StoryObj<typeof SomeComponent>;
export const Default: Story = {

};
```
the "kind" will be "Some/Category/To/SomeComponent" and the "story" will be "Default".

Both of these are combined to the full story name of the format `${kind} ${story}`. This full story name is then used in the include pattern in a filter function, like so:
```
const fullStoryName = `${kind} ${story}`;
const exclude = excludePattern && new RegExp(excludePattern, 'i').test(fullStoryName);
const include = !includePattern || new RegExp(includePattern, 'i').test(fullStoryName);
return !exclude && include;
```
and here are the `excludePattern` and `includePattern` declarations:
```
const excludePattern = options.skipStoriesPattern || configuration.skipStories;
const includePattern = options.storiesFilter || configuration.storiesFilter;
```

Note: If you have a Story object with a name like "SomeStoryHere", spaces will be added between the capital letters, so "story" will be "Some Story Here".
